hudi-common/src/main/java/org/apache/hudi/common/table/log/AbstractHoodieLogRecordReader.java

```diff
@@ -810,13 +817,23 @@ private Option<Function<IndexedRecord, IndexedRecord>> composeEvolvedSchemaTrans
     }
 
     long currentInstantTime = Long.parseLong(dataBlock.getLogBlockHeader().get(INSTANT_TIME));
-    InternalSchema fileSchema = InternalSchemaCache.searchSchemaAndCache(currentInstantTime,
-        hoodieTableMetaClient, false);
+    if (historicalSchemas == null) {
+      FileBasedInternalSchemaStorageManager schemaStorageManager = new FileBasedInternalSchemaStorageManager(hoodieTableMetaClient);
+      historicalSchemas = SerDeHelper.parseSchemas(schemaStorageManager.getHistorySchemaStr());
```

Comment:
cache historical schema，reduce the overhead of search fileSchema.
in our env. we have a log with 1700+ avroBlock,
In original logic， it is very time-consuming to do 1700 fileSchema lookup operations